### ISSUE TITLE
[FR] Theme Styling For More Non-Dashboard Pages and Dialogs

### ISSUE THREAD

**Issue by Nerwyn** [thomasloven/lovelace-card-mod#373]
[The wiki has a page](https://github.com/thomasloven/lovelace-card-mod/wiki/Card-mod-Themes#special-theme-variables) for using card-mod to change the CSS of anything under `hui-root`, `hui-view` (which is already under `hui-root`), and `more-info`, which I've been using to great affect in my theme Material Rounded. I've recently started to implement user customizable Material You style colors using these card-mod special theme variables, but have found that I am not able to alter the CSS of all Home Assistant pages using card-mod. Namely, the developer tools, settings, view configuration, add card, and edit card, ~~and sidebar~~ pages and elements.

Being able to apply card-mod styles to the following HTML elements would help fill in the gaps of what we can theme using card-mod:

- Developer Tools: `ha-panel-developer-tools`
- Settings: `ha-panel-config`
  - Many tools/settings pages use `ha-card` for elements, which do get styled.
- View Configuration: `hui-dialog-edit-view`
- Add Card: `hui-dialog-create-card`
  - The `ha-card` previews do get modified, but not the outer dialog.
- Edit Card: `hui-dialog-edit-card`
- ~~Sidebar: `ha-sidebar`~~
  - `card-mod-sidebar-yaml` is already supported, just wasn't in the wiki.
- Assist: `ha-voice-command-dialog`

Would it also make sense to allow us to apply styles directly to the `home-assistant` HTML element and navigate down using shadow roots instead, if possible?

**Comment by skynet01** [comment]
+1 on this one, I would love to style new headers in the tables right now i cant do it:
![image](https://github.com/thomasloven/lovelace-card-mod/assets/8117544/64c9430e-ca20-4916-af9f-fb34d291edb5)

`skynet_dark_v2:
  card-mod-theme: skynet_dark_v2
  card-mod-view: |
    .group-header {
      font-weight: 800;
      font-size: 18px;
      color: #FFC10A;
    }`

**Comment by Nerwyn** [comment]
It looks like sidebar does have theme variable for the sidebar `card-mod-sidebar-yaml`, it just wasn't in the wiki but is in [this themes README](https://github.com/thomasloven/lovelace-card-mod/blob/f59abc785eabb689ac42a9076197254421f96c60/README-themes.md#theme-variables). That solves a major issue I had with my Material You theming. I wonder if the other patchers can be used for further styling.

**Comment by skynet01** [comment]
I dont think I can target the tables though with the sidebar right?

**Comment by Nerwyn** [comment]
No, but I listed the sidebar as one of the areas I had trouble styling. I've updated the original list.

**Comment by ildar170975** [comment]
Note that `card-mod-sidebar-yaml` is used to mod a sidebar layout of a view, not a "left sidebar with buttons".
Using the `card-mod-sidebar-yaml` is described in the main huge card-mod thread -> 1st post -> link at the bottom -> themes.

**Comment by skynet01** [comment]
anyone?. .. anyone?

**Comment by dcapslock** [comment]
Config pages coming to v4. Dialogs need some more investigation on best way to patch all dialogs with minimal code.

**Comment by dcapslock** [comment]
Dialogs also coming to v4. Will be available for testing in v4.0.0-beta.5

**Comment by dcapslock** [comment]
Dialogs now available in [v4.0.0-beta.5](https://github.com/thomasloven/lovelace-card-mod/releases/tag/v4.0.0-beta.5)

**Comment by dcapslock** [comment]
[v4.0.0](https://github.com/thomasloven/lovelace-card-mod/releases/tag/v4.0.0) released
